okhttp/src/main/java/okhttp3/ResponseBody.java

```diff
@@ -145,24 +145,31 @@ public final InputStream byteStream() {
   /**
    * Returns the response as a character stream decoded with the charset of the Content-Type header.
    * If that header is either absent or lacks a charset, this will attempt to decode the response
-   * body as UTF-8.
+   * body in accordance to its BOM or UTF-8.
    */
-  public final Reader charStream() {
+  public final Reader charStream() throws IOException {
```

Comment:
Source incompatible but binary compatible. Would prefer to defer this work until first interaction of the returned `Reader` but that seems like a lot of machinery.